en/contribute/plugin.md

```diff
@@ -1,9 +1,868 @@
+    console.log('a.b has been changed')
+})
+config.set('a.b.c', 'hello');
+```
+
+**Param**
+
+- `field`: field path，deep watcher must be separated by `.`, such as `user.name`
+- `handler(evt)`: watch handler
+  - `evt.affect(field)` \[Function]:
+    detect whether specific field has been changed
+
+#### - `config.del(field)`
+
+remove some field
+
+**Usage**
+
+```js
+config.del('a.b.c');
+config.get('a.b.c'); //=> undefined
+```
+
+#### - `config.splice(field, start[, delCount[, items...])`
+
+The `Array.prototype.slice`
+
+Except for `field`，params are identical to [Array.prototype.splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
+
+**Example**
+
+```js
+config.set('a.b.c', [1, 2, 3]);
+config.splice('a.b.c', 1, 1);
+config.get('a.b.c'); // => [1,3]
+```
+
+### router
+
+Extending [Routing DSL](../guide/route.md#plugin)
+
+#### - `router.route(register)`
+
+register route，as same as [Routing DSL](../guide/route.md)
+
+**Usage**
+
+```js
+const {route} = router;
+
+route({all, get, post}=>{
+
+  all('/blog').to.send('Hi, Blog')
+  get('/user').to.send('Hi, user')
+  post('/user').to.send({code: 200, data: 'Success!'})
+
+})
+```
+
+**Param**
+
+- register({...methods}):
+  - methods: corresponding to [HTTP methods](../guide/route.md#method)
+
+#### - `router.action(name, builder)`
+
+Register a action , like [`proxy`](../guide/route.md#proxy) or [`json`](../guide/route.md#json)
+
+**Usage**
+
+```js
+const { action } = router;
+action('hello', user => ctx => {
+  ctx.body = `hello ${user}`;
+});
+route(({ all }) => {
+  all('/blog').hello('svrx'); //=> response 'hello svrx'
+});
+```
+
+**Param**
+
+- `name` \[String]: action name
+- `builder(payload)`
+  - payload: payload that passed to action，like `'svrx'` in above example
+  - Return: builder returns 的[koa style](https://koajs.com) middleware
+
+#### - `router.load(filename)`
+
+Load routing file manually ，Same as [options `--route`](../guide/route#start)
+
+_Also support hot reloading_
+
+**Usage**
+
+```js
+await router.load('/path/to/route.md');
+```
+
+**Param**
+
+- `filename`: absolute path for routing file
+
+**Return**
+
+`Promise`
+
+### logger
+
+Logger module, who's level can be controlled by `logger.level` (default is `warn`)
+
+```js
+svrx({
+  logger: {
+    level: 'error'
+  }
+});
+```
+
+or cli way
+
+```bash
+svrx --logger.level error
+```
+
+Above example will output log that more than `warn`, such as `notify`, `error`
+
+### - `logger[level](msg)`
+
+Svrx provides multiple levels of logging: `slient`, `notify`, `error`, `warn` (default), `info`, `debug`
+
+**Usage**
+
+```js
+logger.notify('notify'); // show `nofity`
+logger.error('error'); // show `error` and `notify`
+logger.warn('warn'); // show `warn`、`error` and `norify`
+```
+
+> `logger.log` is an alias for `logger.notify`
+
+### io
+
+io used for the communication between server and client ， Please check it with [client-side io](#client-io)
```

Comment:
io is used for the communication between server and client. Please check it out in [client-side io](#client-io)